source/fundamentals/builders/indexes.txt

```diff
@@ -11,3 +11,165 @@ Indexes Builders
+method. Then, pass the BSON instance into the createIndex() method to
+create a descending index. 
+
+A descending index sorts the index key from largest to smallest.
+
+The following example specifies a descending index on the ``capacity`` field: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/Indexes.java
+   :language: java
+   :dedent:
+   :start-after: begin descendingIndex
+   :end-before: end descendingIndex
+
+.. _index-compound-indexes:
+
+Compound Indexes
+----------------
+
+Create a BSON instance by passing the field names you want to index to
+the :java-core-api:`compoundIndex() <com/mongodb/client/model/Indexes.html#compoundIndex(org.bson.conversions.Bson...)>`
+method. Then, pass the BSON instance into the createIndex() method to
+create a compound index. 
+
+The following example specifies a compound index composed of
+descending index on the ``capacity`` and ``year`` field, followed
+by an ascending index on the ``name`` field: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/Indexes.java
+   :language: java
+   :dedent:
+   :start-after: begin compoundIndexExample
+   :end-before: end compoundIndexExample
+
+.. _index-text-indexes:
+
+Text Indexes
+------------
+
+Create a BSON instance by passing the field name you want to index to
+the :java-core-api:`text() <com/mongodb/client/model/Indexes.html#text(java.lang.String)>`
+method. Then, pass the BSON instance into the createIndex() method to
+create a text index key. 
+
+A text index groups the index key by the word you specify.
+
+The following example specifies a text index key on the "theaters" field:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/Indexes.java
+   :language: java
+   :dedent:
+   :start-after: begin textIndex
+   :end-before: end textIndex
+
+.. _index-hashed-indexes:
+
+Hashed Indexes
+--------------
+
+Create a BSON instance by passing the field name you want to index to
+the :java-core-api:`hashed() <com/mongodb/client/model/Indexes.html#hashed(java.lang.String)>`
+method. Then, pass the BSON instance into the createIndex() method to
+create a hashed index key. 
+
+A hashed index groups the index key by its hash value.
+
+The following example specifies a hashed index on the ``capacity``
+field:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/Indexes.java
+   :language: java
+   :dedent:
+   :start-after: begin hashedIndex
+   :end-before: end hashedIndex
+
+.. _index-geospatial-indexes:
+
+Geospatial Indexes
+------------------
+
+Create a BSON instance by passing the field name or names you want to
+index to the :java-core-api:`geo2dsphere() <com/mongodb/client/model/Indexes.html#geo2dsphere(java.lang.String...)>`
+method. Then, pass the BSON instance into the createIndex() method to
+create a geo2dsphere index key. 
+
+A geo2dsphere index groups the index key by their coordinates.
```

Comment:
This might be more accurate and succinct since the index groups documents, not index keys:

```suggestion
A geo2dsphere index groups documents by the coordinates in the indexed field.
```

Similar comment applies to all instances in which you mention "groups the index key".